Makefile

```diff
@@ -334,11 +336,19 @@ undeploy_bundle:
 create_cryostat_cr: destroy_cryostat_cr
 	$(CLUSTER_CLIENT) create -f config/samples/operator_v1beta1_cryostat.yaml
 
+.PHONY: create_clustercryostat_cr
+create_clustercryostat_cr: destroy_clustercryostat_cr
+	$(CLUSTER_CLIENT) patch -f config/samples/operator_v1beta1_clustercryostat.yaml --local=true --type=merge -p '{"spec": {"installNamespace": "$(DEPLOY_NAMESPACE)", "targetNamespaces": ["$(DEPLOY_NAMESPACE)"]}}' -o yaml | oc apply -f -
```

Comment:
There must be a nicer way to do this, but I modified this:

```diff
diff --git a/Makefile b/Makefile
index 36798a2..63d8301 100644
--- a/Makefile
+++ b/Makefile
@@ -89,6 +89,7 @@ CUSTOM_SCORECARD_VERSION ?= 2.3.0-$(shell date -u '+%Y%m%d%H%M%S')
 export CUSTOM_SCORECARD_IMG ?= $(IMAGE_TAG_BASE)-scorecard:$(CUSTOM_SCORECARD_VERSION)
 
 DEPLOY_NAMESPACE ?= cryostat-operator-system
+TARGET_NAMESPACES ?= "[$(DEPLOY_NAMESPACE)]"
 SCORECARD_NAMESPACE ?= cryostat-operator-scorecard
 
 # Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
@@ -338,7 +339,7 @@ create_cryostat_cr: destroy_cryostat_cr
 
 .PHONY: create_clustercryostat_cr
 create_clustercryostat_cr: destroy_clustercryostat_cr
-       $(CLUSTER_CLIENT) patch -f config/samples/operator_v1beta1_clustercryostat.yaml --local=true --type=merge -p '{"spec": {"installNamespace": "$(DEPLOY_NAMESPACE)", "targetNamespaces": ["$(DEPLOY_NAMESPACE)"]}}' -o yaml | oc apply -f -
+       $(CLUSTER_CLIENT) patch -f config/samples/operator_v1beta1_clustercryostat.yaml --local=true --type=merge -p '{"spec": {"installNamespace": "$(DEPLOY_NAMESPACE)", "targetNamespaces": $(TARGET_NAMESPACES)}}' -o yaml | oc apply -f -
 
 # Undeploy a Cryostat instance
 .PHONY: destroy_cryostat_cr
```

and invoked it like this:

```bash
$ make DEPLOY_NAMESPACE=$(oc project -q) TARGET_NAMESPACES=[\"test-apps\",\"cryostat-test\"] create_clustercryostat_cr
```

so I could test a clustercryostat that has more than just its own installation namespace to watch.

Do you think some neater version of this patch is worthwhile?

Reply:
![image](https://user-images.githubusercontent.com/3787464/226405061-57e8021c-6782-4199-bac9-6a98d1bf9411.png)

```
$ oc get all -n cryostat-test
NAME                                         READY   STATUS    RESTARTS   AGE
pod/clustercryostat-sample-cb87d985f-t6z77   3/3     Running   0          4m51s

NAME                                     TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)             AGE
service/clustercryostat-sample           ClusterIP   10.217.4.227   <none>        8181/TCP,9091/TCP   4m51s
service/clustercryostat-sample-grafana   ClusterIP   10.217.4.173   <none>        3000/TCP            4m52s

NAME                                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/clustercryostat-sample   1/1     1            1           4m51s

NAME                                               DESIRED   CURRENT   READY   AGE
replicaset.apps/clustercryostat-sample-cb87d985f   1         1         1       4m51s

NAME                                                      HOST/PORT                                                       PATH   SERVICES                         PORT   TERMINATION   WILDCARD
route.route.openshift.io/clustercryostat-sample           clustercryostat-sample-cryostat-test.apps-crc.testing                  clustercryostat-sample           8181   reencrypt     None
route.route.openshift.io/clustercryostat-sample-grafana   clustercryostat-sample-grafana-cryostat-test.apps-crc.testing          clustercryostat-sample-grafana   3000   reencrypt     None
```

----

```yaml
apiVersion: v1
items:
- apiVersion: operator.cryostat.io/v1beta1
  kind: ClusterCryostat
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"operator.cryostat.io/v1beta1","kind":"ClusterCryostat","metadata":{"annotations":{},"name":"clustercryostat-sample"},"spec":{"enableCertManager":true,"eventTemplates":[],"installNamespace":"cryostat-test","minimal":false,"reportOptions":{"replicas":0},"storageOptions":{"pvc":{"annotations":{},"labels":{},"spec":{}}},"targetNamespaces":["test-apps","cryostat-test"],"trustedCertSecrets":[]}}
    creationTimestamp: "2023-03-20T16:20:57Z"
    finalizers:
    - operator.cryostat.io/cryostat.finalizer
    generation: 2
    name: clustercryostat-sample
    resourceVersion: "46031"
    uid: 07d8f65e-462c-4bd1-8728-cf6de7795696
  spec:
    enableCertManager: true
    installNamespace: cryostat-test
    minimal: false
    reportOptions:
      resources: {}
    storageOptions:
      pvc:
        spec:
          resources: {}
    targetNamespaces:
    - test-apps
    - cryostat-test
  status:
    applicationUrl: https://clustercryostat-sample-cryostat-test.apps-crc.testing
    conditions:
    - lastTransitionTime: "2023-03-20T16:21:02Z"
      message: All certificates for Cryostat components are ready.
      reason: AllCertificatesReady
      status: "True"
      type: TLSSetupComplete
    - lastTransitionTime: "2023-03-20T16:21:03Z"
      message: ReplicaSet "clustercryostat-sample-cb87d985f" has successfully progressed.
      reason: NewReplicaSetAvailable
      status: "True"
      type: MainDeploymentProgressing
    - lastTransitionTime: "2023-03-20T16:21:15Z"
      message: Deployment has minimum availability.
      reason: MinimumReplicasAvailable
      status: "True"
      type: MainDeploymentAvailable
    grafanaSecret: clustercryostat-sample-grafana-basic
    targetNamespaces:
    - test-apps
    - cryostat-test
kind: List
metadata:
  resourceVersion: ""
```

----

```
$ oc get all -n test-apps
NAME                                READY   STATUS    RESTARTS   AGE
pod/quarkus-test-68b7769c98-8h64h   1/1     Running   0          76s
pod/quarkus-test-68b7769c98-98mqv   1/1     Running   0          76s
pod/quarkus-test-68b7769c98-wcvwv   1/1     Running   0          100s

NAME                   TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)                                AGE
service/quarkus-test   ClusterIP   10.217.4.70   <none>        8080/TCP,8443/TCP,9097/TCP,10010/TCP   101s

NAME                           READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/quarkus-test   3/3     3            3           101s

NAME                                      DESIRED   CURRENT   READY   AGE
replicaset.apps/quarkus-test-5665f6dc54   0         0         0       101s
replicaset.apps/quarkus-test-68b7769c98   3         3         3       101s

NAME                                          IMAGE REPOSITORY                                                                 TAGS     UPDATED
imagestream.image.openshift.io/quarkus-test   default-route-openshift-image-registry.apps-crc.testing/test-apps/quarkus-test   0.0.10   About a minute ago
```

Reply:
As a side note:

![image](https://user-images.githubusercontent.com/3787464/226405487-7bd46fd1-8f5a-48ae-bbb8-07b818609ba3.png)

:D

Reply:
I think that's fair enough, in this instance the user is probably a contributor/developer so adding basic tooling like that shouldn't be a problem.

Reply:
I added something now where `TARGET_NAMESPACES` is space separated and translated into a JSON array with jq. How does it look?